lib/patch-https-request-for-proxying.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Monkey-patch https.request to proxy request to the broker server
+ * according the the env variables `https_proxy` and `no_proxy`:
+ * get the https proxy from the env. proxy requests unless the
+ * broker server is in the no_proxy list.
+ * Start from the "Entry point" below.
+ */
+const url = require('url');
+const tunnel = require('tunnel');
+const {brokerServerUrl} = require('./config');
+const brokerServer = url.parse(brokerServerUrl || '');
```

Comment:
this is what `lib/config` does https://github.com/snyk/broker/blob/master/lib/config/index.js#L8